src/command/schema/push.rs

```diff
@@ -0,0 +1,70 @@
+use crate::client::get_rover_client;
+use anyhow::Result;
+use rover_client::query::schema::push;
+use std::path::{Path, PathBuf};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct Push {
+    /// where to find the schema. .graphql, .json or uri
+    #[structopt(name = "SCHEMA_PATH", parse(from_os_str))]
+    schema_path: PathBuf,
+    /// The variant of the request graph from Apollo Studio
+    #[structopt(long, default_value = "current")]
+    variant: String,
+    #[structopt(long)]
+    graph: String,
+    #[structopt(long, default_value = "default")]
+    profile: String,
+}
+
+impl Push {
+    pub fn run(&self) -> Result<()> {
+        let client = get_rover_client(&self.profile)?;
+        log::info!(
+            "Let's push this schema, {}@{}, mx. {}!",
+            &self.graph,
+            &self.variant,
+            &self.profile
+        );
+
+        let schema_document = get_schema_from_file_path(&self.schema_path)?;
+
+        let push_response = push::run(
+            push::push_schema_mutation::Variables {
+                graph_id: self.graph.clone(),
+                variant: self.variant.clone(),
+                schema_document: Some(schema_document),
+            },
+            client,
+        )?;
+
+        log::info!(
+            "{}\nSchema Hash: {}",
+            push_response.message,
+            push_response.schema_hash
+        );
+        Ok(())
+    }
+}
+
+fn get_schema_from_file_path(path: &PathBuf) -> Result<String> {
+    if Path::exists(path) {
+        let contents = std::fs::read_to_string(path)?;
+        Ok(contents)
+    } else {
+        Err(anyhow::anyhow!(
+            "Invalid path. No file found at {}",
+            path.display()
+        ))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    #[test]
+    fn get_schema_from_file_path_loads() {
+        // todo @jake -- add test for this after merging with avery's work
```

Comment:
what is blocking this?

Reply:
It was the temp file stuff you had written for one of your PRs

Reply:
ooh gotcha - i actually removed that in favor of using `assert_fs`, which is now in `main`. here's a few examples of using that library

https://github.com/apollographql/rover/blob/7b2dab780fb9c728f16730d751a5786961e6e150/src/command/config/api_key.rs#L58
https://github.com/apollographql/rover/pull/52/files#diff-6fcf03b28a7c3fdfc4ed29ed75e7fc1de9689f4e391b6443e5bd7de817c1d7e9R72